model/implementation/DeliveryService.php

```diff
@@ -241,41 +347,154 @@ public function unassignTestTaker($testTakerId, $deliveryId)
     }
 
     /**
+     * Sets a delivery execution in the awaiting state
+     *
+     * @param string $executionId
+     * @return bool
+     */
+    public function waitExecution($executionId)
+    {
+        $deliveryExecution = $this->getDeliveryExecution($executionId);
+        $executionState = $this->getState($deliveryExecution);
+        $result = false;
+
+        if (self::STATE_ACTIVE == $executionState || self::STATE_PAUSED == $executionState) {
+            $this->setProctoringState($deliveryExecution, self::STATE_AWAITING);
+
+            $result = true;
+        }
+
+        return $result;
+    }
+
+    /**
+     * Sets a delivery execution in the inprogress state
+     *
+     * @param string $executionId
+     * @return bool
+     */
+    public function processExecution($executionId)
+    {
+        $deliveryExecution = $this->getDeliveryExecution($executionId);
+        $executionState = $this->getState($deliveryExecution);
+        $result = false;
+
+        if (self::STATE_AUTHORIZED == $executionState) {
+            $this->setProctoringState($deliveryExecution, self::STATE_INPROGRESS);
+
+            $session = $this->getTestSession($deliveryExecution);
+            if ($session->getState() == AssessmentTestSessionState::SUSPENDED) {
+                $session->resume();
+                $this->getStorage()->persist($session);
+            }
+
+            $result = true;
+        }
+
+        return $result;
+    }
+
+    /**
      * Authorises a delivery execution
      *
      * @param string $executionId
+     * @param array $reason
      * @return bool
      */
-    public function authoriseExecution($executionId)
+    public function authoriseExecution($executionId, $reason = null)
```

Comment:
small notworthy details : in some part of the code we use the British English "authorise" and in some other part we use the American one "authorize". It would be nice to switch to the american one as a coding standard maybe ?
(I suggest doing it in another PR, this one is too huge already and we may break things in the process)